language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/GetterCodeAction.java

```diff
@@ -0,0 +1,71 @@
+package org.ballerinalang.langserver.codeaction.providers;
+
+import io.ballerina.compiler.syntax.tree.ClassDefinitionNode;
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.ObjectFieldNode;
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.codeaction.CodeActionUtil;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.CodeActionContext;
+import org.ballerinalang.langserver.commons.codeaction.CodeActionNodeType;
+import org.ballerinalang.langserver.commons.codeaction.spi.NodeBasedPositionDetails;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Code Action for getters.
+ *
+ * @since 2201.1.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class GetterCodeAction extends AbstractCodeActionProvider {
+
+    public static final String NAME = "Getter";
+    public GetterCodeAction() {
+        super(Arrays.asList(CodeActionNodeType.OBJECT_FIELD));
+    }
+
+    @Override
+    public List<CodeAction> getNodeBasedCodeActions(CodeActionContext context,
+                                                    NodeBasedPositionDetails posDetails) {
+
+        if (!context.currentSyntaxTree().isPresent()) {
+            return Collections.emptyList();
+        }
+
+        NonTerminalNode matchedNode = CommonUtil.findNode(new Range(context.cursorPosition(), context.cursorPosition()),
+                context.currentSyntaxTree().get());
+        String commandTitle = "Create a getter for " + ((ObjectFieldNode) matchedNode).fieldName().toString();
```

Comment:
Shall we add inverted commas to the fieldName? And we can add `"Create a getter for %s" to the `CommandConstants.java`

Reply:
Added in 85613cc5bfa90a74b38fb3e48f237c79dde35307